compiler/src/dotty/tools/dotc/sbt/ExtractAPI.scala

```diff
@@ -140,9 +142,19 @@ private class ExtractAPICollector(using Context) extends ThunkHolder {
   /** This cache is necessary to avoid unstable name hashing when `typeCache` is present,
    *  see the comment in the `RefinedType` case in `computeType`
    *  The cache key is (api of RefinedType#parent, api of RefinedType#refinedInfo).
-    */
+   */
   private val refinedTypeCache = new mutable.HashMap[(api.Type, api.Definition), api.Structure]
 
+  /** This cache is necessary to avoid infinite loops when hashing an inline "Body" annotation.
+   *  Its values are transitively seen inline references within a call chain starting from a single "origin" inline
+   *  definition. Avoid hashing an inline "Body" annotation if its associated definition is already in the cache.
+   *  Precondition: the cache is empty whenever we hash a new "origin" inline "Body" annotation.
+   */
+  private val seenInlineCache = mutable.HashSet.empty[Symbol]
+
+  /** This cache is optional, it avoids recomputing hashes of inline "Body" annotations. */
+  private val inlineBodyCache = mutable.HashMap.empty[Symbol, Int]
```

Comment:
In which  situation is this cache used? Is this to address the FIXME in apiAnnotations?

Reply:
when you inherit an inline method it runs the hashing again

Reply:
I added lines to both this doc comment and the fixme about this

Reply:
now deleted the fixme